docs/development/using.md

```diff
@@ -50,6 +50,50 @@ migration.
 restart if it crashes, and `upgrade-zulip` will take care of running
 migrations and then cleanly restaring the server for you).
 
+## Accessing the Vagrant development environment from other systems
+
+If you want to access your Vagrant development environment from outside of it,
+then you'll need to follow a few extra steps.
+
+This required when dealing with testing third-party services like Jira and
+Bitbucket Server which can be hosted locally if you want to allow these
+services to be able to directly post to your Zulip development enviroment.
+Or if you're testing the mobile app and need to use a locally hosted
+development version of the zulip server.
+
+To do either one of these you'll need to make sure that your Zulip development
+environment (notably the Vagrant virtual machine) is accessible externally.
+
+### Step One: Get the Vagrant VM to listen on all available addresses.
+Open zulip/Vagrantfile. Replace the line `host_ip_addr = "127.0.0.1"` with
+`host_ip_addr = "0.0.0.0"`. This means that the VM is now listening on all IPv4
+addresses available on your local machine. This will allow other services like
+Jira or Bitbucket Server to now communicate with Zulip running on the
+development server.
+
+### Step Two: Figure out your IP address.
+If you're using Linux, this can be found by running `ifconfig`, or
+`ipconfig /all` if you're on Windows, outside of the Zulip development
+environment (on your local PC). In the output you should see a line resembling
+`inet 192.168.1.7`, in this case `192.168.1.7` would be your IP address.
+
+### Step Three: Set EXTERNAL_HOST.
+Like most complex web apps, the Zulip server has an idea internally of what
+base URL it's supposed to be accessed at; we call this setting `EXTERNAL_HOST`.
+In development, the setting is normally `localhost:9991`, and corresponds to a
+base URL of `http://localhost:9991/`.
+
+Set this to `IPADDRESS:9991`, where `IPADDRESS` is the address you identified
+in the previous step. In development, we can do this with an environment
+variable. For example, if your IP address is `192.168.1.7` then the environment
+variable can be set using the command: `export EXTERNAL_HOST="192.168.1.7:9991"`
+
+### Step Four: Running the development environment.
+Now, from the vagrant development environment, run the server by passing
+`--interfaces=''` to `./tools/run-dev.py` command. That is, run
```

Comment:
I think we can be a lot more brief, and I preferred the approach that uses `env EXTERNAL_HOST`, because it's not modal (so you can just copy-paste it freely).

Reply:
@timabbott, for being more brief, should I remove lines like 
> This means that the VM is now listening on all IPv4
addresses available on your local machine. This will allow other services like
Jira or Bitbucket Server to now communicate with Zulip running on the
development server.

Reply:
I could remove all lines where we're doing some extra explaining and make it more of a "copy-paste to get the job done" kinda guide.

Reply:
No, those explanations are great and we want to keep.  I just mean repeating the `--interfaces=''` part on two consecutive lines.